CONTRIBUTING.md

```diff
@@ -1,35 +1,211 @@
+
+## What is and actual bug report?
+
+Developers and contributors are not supposed to deal with issues for which little to no investigation to find the actual cause of a purported issue was made by the reporter.
+
+Positive contributions are those which are reported with efforts to find the actual cause of an issue, or at the very least efforts were made to narrow it as much as possible.
+
+Requiring people to investigate as much as possible before opening an issue will more than likely avoid burdening the project with invalid issues or issues unrelated to qBittorrent.
+
+The following are _not_ bug reports; **Check the wiki, forum or other places for help and support for issues like these**:
+
+-   Explanation of qBittorrent options (see wiki);
+-   Help with Web UI setup;
+-   Help with embedded tracker setup;
+-   Help about Bittorrent in general;
+-   Issues with specific search plugins;
+-   Asking for specific builds of qBittorrent not currently officially available;
+    -   If you really need one such older unavailable build, compile it yourself from source (see wiki) or get someone to do it for you on the forum, for example.
+-   Possibly others; read on and use common sense.
+
+
+
+The issue tracker is for provable issues only: You will have to make the case that the issue is really with qBittorrent and not something else on your side.
+
+To make a case means to provide detailed steps so that anybody can reproduce the issue.
+Be sure to rule out that the issue is not caused by something specific on your side.
+
+Any issue opened without effort to provide the required details for developers, contributors or anybody else to reproduce the problem will be closed as invalid.
+For example:
+-   Crash reports with just a stack trace;
+-   Speculated performance issues that do not come with actual profiling data + analysis supporting the claim;
+
+## Before submitting a bug report
+
+-   **Do some basic troubleshooting**:
+    -   restart qBittorrent;
+    -   restart your PC;
+    -   update your OS (e.g. Windows updates);
+    -   fully reinstall qBittorrent;
+    -   etc...
+-   Make sure the problem is not caused by anti-virus or other program messing with your files;
+-   Check if you can reproduce the problem in the latest version of qBittorrent;
+-   **Check forum and wiki**. You might be able to find the cause of the problem and fix things yourself;
+-   **Check if the issue exists already in the issue tracker**.
+    -   If it does and the issue is still open, add a comment to the existing issue instead of opening a new one;
+    -   If you find a Closed issue that seems like it is the same thing that you're experiencing, open a new issue and include a link to the original issue in the body of your new one.
+-   If the issue is with the search functionality:
+    -   make sure you have `python` installed correctly (remember the search functionality requires a working python installation);
```

Comment:
python word as a link

Reply:
And capitalize

Reply:
Ok

Reply:
Capitalize the first word, not the sentence.

Reply:
Ok